sklearn/manifold/isomap.py

```diff
@@ -109,13 +202,29 @@ def _fit_transform(self, X):
         kng = kneighbors_graph(self.nbrs_, self.n_neighbors,
                                mode='distance')
 
-        self.dist_matrix_ = graph_shortest_path(kng,
-                                                method=self.path_method,
-                                                directed=False)
-        G = self.dist_matrix_ ** 2
-        G *= -0.5
+        self.dist_matrix_ = shortest_path(kng,
+                                          method='D',
+                                          directed=False,
+                                          indices=landmarks).T
```

Comment:
Only pass ``landmarks`` if necessary (as currently written, it's slightly slower for the default case)